fs/entry.go

```diff
@@ -91,8 +91,15 @@ func IterateEntries(ctx context.Context, dir Directory, cb func(context.Context,
 }
 
 // DirectoryIterator iterates entries in a directory.
+//
+// The client is expected to call Next() in a loop until it returns a nil entry to signal
+// end of iteration or until an error has occurred.
```

Comment:
Do we also want to clarify that code expects either (non-nil entry, error, end of iterator) since much of the code loops on `e := Next(); e != nil` without checking the error value. I'm not really sure under what conditions you could end up with a non-nil entry and error, but it could cause problems if that happens

Reply:
Added comment to clarify valid results:

```
(nil,nil) - end of iteration, success
(entry,nil) - iteration in progress, success
(nil,err) - iteration stopped, failure
```

Reply:
Can we explicitly add this to the function documentation?